components/ui/Modal/Modal.tsx

```diff
@@ -7,45 +7,53 @@ import {
   enableBodyScroll,
   clearAllBodyScrollLocks,
 } from 'body-scroll-lock'
-
+import FocusTrap from '@lib/focus-trap'
 interface Props {
   className?: string
   children?: any
   open?: boolean
   onClose: () => void
+  onEnter?: () => void | null
 }
 
-const Modal: FC<Props> = ({ children, open, onClose }) => {
+const Modal: FC<Props> = ({ children, open, onClose, onEnter = null }) => {
   const ref = useRef() as React.MutableRefObject<HTMLDivElement>
 
+  const handleKey = (e: KeyboardEvent) =>
+    useCallback(() => {
+      if (e.key === 'Escape') {
+        return onClose()
+      }
+    }, [onClose])
+
   useEffect(() => {
     if (ref.current) {
       if (open) {
         disableBodyScroll(ref.current)
+        window.addEventListener('keydown', handleKey)
       } else {
         enableBodyScroll(ref.current)
       }
     }
     return () => {
+      window.removeEventListener('keydown', handleKey)
       clearAllBodyScrollLocks()
     }
   }, [open])
```

Comment:
```suggestion
  }, [open, handleKey])
```

`useCallback` was added to `handleKey` with the intention of adding it as a dep to this `useEffect`.

An alternative is you could move the `handleKey` function to be inside the `useEffect` and then add `onClose` as a dependency.